tests/lib/prepare.sh

```diff
@@ -145,5 +148,9 @@ prepare_all_snap() {
         fi
     fi
 
+    # ensure no test creates havoc
+    for name in pc pc-kernel ubuntu-core; do
+        snap list | grep $name
```

Comment:
These failures are tough to debug without being familiar with the tests, for example:

```
Removing a snap try dir does not break everything

+ rm -rf /home/gopath/src/github.com/snapcore/snapd/tests/main/try-snap-goes-away/trydir
+ grep core
+ snap list
```

Took me a while to figure out what exactly happened. It might be helpful to catch the failure and print something pointing in the right direction (even as simple as "Expected snap '$name' to be installed").